horovod/common/operations.cc

```diff
@@ -1480,7 +1551,11 @@ void BackgroundThreadLoop(HorovodGlobalState& state) {
 
   MPI_Comm_free(&horovod_global.mpi_comm);
 
+#if HAVE_DDL
+  ddl_finalize();
+#else
   MPI_Finalize();
```

Comment:
Does `ddl_finalize()` do `MPI_Finalize()` as well?

Reply:
Yes and also cleanup some MPI communicators so we cannot call MPI_Finalize before.